File: python/exercise_121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buyingPrice = prices[0]
        sellingPrice = prices[0]
        stonks = 0
        for i in range(len(prices)):
            if prices[i] < buyingPrice:
                buyingPrice = prices[i]
                sellingPrice = prices[i]
            if prices[i] > sellingPrice:
                if prices[i]-buyingPrice>stonks:
                    sellingPrice = prices[i]
                    stonks = prices[i]-buyingPrice
        return  stonks

File: java/arrays/exercise_121.java
class Solution {
    public int maxProfit(int[] prices) {
        int buyingPrice = prices[0];
        int sellingPrice = prices[0];
        int stonks = 0;
        for (int i=0;i<prices.length;i++){
            if (prices[i] < buyingPrice){
                buyingPrice = prices[i];
                sellingPrice = prices[i];
            }
            if (prices[i] > sellingPrice){
                if (prices[i]-buyingPrice>stonks){
                    sellingPrice = prices[i];
                    stonks = prices[i]-buyingPrice;
                }
            }
        }
        return stonks;
    }
}